From 7b4abf83417beeaf6cc178303d1f118017a48167 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jo=C3=A3o=20Valverde?= <j@v6e.pt>
Date: Sat, 8 Oct 2022 16:47:51 +0100
Subject: [PATCH] dfilter: Fix integer comparison on big-endian

Fix copy-paste mistake to use the correct struct field.

Fixes #12236.
---
 epan/ftypes/ftype-integer.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/epan/ftypes/ftype-integer.c b/epan/ftypes/ftype-integer.c
index b611fd71a2..833ec0bfa7 100644
--- a/epan/ftypes/ftype-integer.c
+++ b/epan/ftypes/ftype-integer.c
@@ -1113,10 +1113,7 @@ static enum ft_result sint_val_to_sinteger64(const fvalue_t *src, gint64 *dst)
 
 static enum ft_result uint64_val_to_uinteger64(const fvalue_t *src, guint64 *dst)
 {
-	if (src->value.sinteger < 0)
-		return FT_OVERFLOW;
-
-	*dst = (guint64)src->value.sinteger;
+	*dst = src->value.uinteger64;
 	return FT_OK;
 }
 
-- 
GitLab

